manager/volume.go

```diff
@@ -501,6 +499,61 @@ func (m *VolumeManager) Activate(volumeName string, frontend string) (v *longhor
 	return v, nil
 }
 
+func (m *VolumeManager) syncBackupAndRestoreWithLastBackup(volume *longhorn.Volume, engine *longhorn.Engine) (bool, error) {
+	// trigger BackupVolume synchronization
+	var backupVolumeName string
+	if volume.Status.RestoreRequired {
+		name, ok := volume.Labels[types.LonghornLabelBackupVolume]
+		if !ok {
+			return false, errors.New("Cannot find the backup volume label")
+		}
+		backupVolumeName = name
+	} else {
+		backupVolumeName = volume.Name
+	}
+
+	backupVolume, err := m.ds.GetBackupVolume(backupVolumeName)
+	if err != nil {
+		return false, errors.Wrapf(err, "failed to get backup volume: %v", backupVolumeName)
+	}
+	requestSyncTime := metav1.Time{Time: time.Now().UTC()}
+	backupVolume.Spec.SyncRequestedAt = requestSyncTime
+	if _, err = m.ds.UpdateBackupVolume(backupVolume); err != nil {
+		return false, errors.Wrapf(err, "failed to update backup volume:%v", backupVolumeName)
+	}
+
+	// confirm BackupVolume is synchronized
+	for i := 0; i < datastore.VerificationRetryCounts; i++ {
+		backupVolume, err = m.ds.GetBackupVolumeRO(backupVolumeName)
+		if err != nil {
+			return false, errors.Wrapf(err, "failed to get backup volume: %v", volume.Name)
+		}
+
+		if backupVolume.Status.LastSyncedAt.After(backupVolume.Spec.SyncRequestedAt.Time) ||
+			backupVolume.Status.LastSyncedAt.Equal(backupVolume.Spec.SyncRequestedAt.DeepCopy()) {
+			break
+		}
+		time.Sleep(datastore.VerificationRetryInterval)
+	}
```

Comment:
Retrying here looks a bit weird.

Reply:
Because I want to make sure I get the correct latest backup volume name after the backup volume controller reconciles, but don't want to wait a fixed amount of time. Therefore, I use a loop to repeat the attempts to check if the sync is complete.

Reply:
Does it cause any problems if the backup volume is not synchronized in the given period of time?

Reply:
If the backup synchronization fails and the retry time is exceeded, the activated disaster recovery volume will not have the latest data like the current version. 

So if syncing the backup volume fails, I will return an error message.